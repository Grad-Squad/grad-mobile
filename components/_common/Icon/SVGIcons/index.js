import SVGIcons from './SVGIcons';

export default SVGIcons;
