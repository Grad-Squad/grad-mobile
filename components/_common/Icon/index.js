export { default as Icon } from './Icon';
export { default as PressableIcon } from './PressableIcon';
export { default as MaterialTypeIconsMap } from './MaterialTypeIconsMap';
