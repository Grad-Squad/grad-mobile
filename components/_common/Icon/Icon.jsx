import React from 'react';
import PropTypes from 'prop-types';
import { Colors } from 'styles';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import Feather from 'react-native-vector-icons/Feather';
import AntDesign from 'react-native-vector-icons/AntDesign';
import { TextPropType } from 'proptypes';
import SVGIcons from './SVGIcons';

const map = {
  question: AntDesign,
  close: Ionicons,
  'home-outline': Ionicons,
  'radio-button-checked': MaterialIcons,
  'radio-button-off': MaterialIcons,
  'bell-outline': MaterialCommunityIcons,
  search: Octicons,
  'add-circle-outline': MaterialIcons,
  bookmark: Feather,
  menu: Feather,
  'cards-outline': MaterialCommunityIcons,
  'format-list-checkbox': MaterialCommunityIcons,
  pdffile1: AntDesign,
  'image-outline': MaterialCommunityIcons,
  'ondemand-video': MaterialIcons,
  'dots-horizontal': MaterialCommunityIcons,
  'dots-vertical': MaterialCommunityIcons,
  'edit-2': Feather,
  'check-square': Feather,
  square: Feather,
  AddImage: SVGIcons,
  'delete-outline': MaterialIcons,
  'send':MaterialIcons,
  "arrow-left":MaterialCommunityIcons,
  'plus':Feather,
  'comment-outline':MaterialCommunityIcons,
};

const Icon = ({ name, size, color, style }) => {
  const IconLibrary = map[name] || map.question;
  return <IconLibrary name={name} size={size} color={color} style={style} />;
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number,
  color: PropTypes.string,
  style: TextPropType,
};
Icon.defaultProps = {
  size: 26,
  color: Colors.offBlack,
  style: {},
};

export default React.memo(Icon);
