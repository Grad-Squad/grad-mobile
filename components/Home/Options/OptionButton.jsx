import React from 'react';
import PropTypes from 'prop-types';
import { Pressable, StyleSheet } from 'react-native';
import { IconNames } from 'common/Icon/Icon';
import EduText from 'common/EduText';
import pressableAndroidRipple from 'common/pressableAndroidRipple';
import { Icon } from 'common/Icon';

const OptionButton = ({ label, iconName, onPress }) => (
  <Pressable
    android_ripple={pressableAndroidRipple}
    onPress={onPress}
    style={styles.optionButton}
  >
    <Icon name={iconName} size={30} />
    <EduText style={styles.optionName}>{label}</EduText>
  </Pressable>
);

OptionButton.propTypes = {
  label: PropTypes.string.isRequired,
  iconName: PropTypes.oneOf(Object.values(IconNames)),
  onPress: PropTypes.func.isRequired,
};
OptionButton.defaultProps = { iconName: IconNames.settings };

export default OptionButton;

const styles = StyleSheet.create({
  optionButton: {
    flexDirection: 'row',
    alignItems: 'center',

    marginVertical: 5,
    paddingVertical: 5,
    paddingHorizontal: 10,
  },
  optionName: {
    fontSize: 25,

    marginLeft: 5,
  },
});
