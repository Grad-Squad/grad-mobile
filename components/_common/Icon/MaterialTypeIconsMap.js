import { IconNames } from "./Icon";

export default {
  Flashcards: IconNames.cards,
  MCQ: IconNames.checklist,
  PDF: IconNames.pdf,
  Images: IconNames.image,
  Video: IconNames.video,
};
