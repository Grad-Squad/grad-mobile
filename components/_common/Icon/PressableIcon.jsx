import React from 'react';
import PropTypes from 'prop-types';
import { Pressable } from 'react-native';
import pressableAndroidRipple from 'common/pressableAndroidRipple';
import { stylePropType } from 'proptypes';
import Icon from './Icon';

const PressableIcon = ({ onPress, style, pressableProps, ...rest }) => (
  <Pressable
    onPress={onPress}
    android_ripple={pressableAndroidRipple}
    style={style}
    // eslint-disable-next-line react/jsx-props-no-spreading
    {...pressableProps}
  >
    <Icon {...rest} />
  </Pressable>
);

PressableIcon.propTypes = {
  onPress: PropTypes.func.isRequired,
  style: stylePropType,
  // eslint-disable-next-line react/forbid-prop-types
  pressableProps: PropTypes.object,
};
PressableIcon.defaultProps = {
  style: {},
  pressableProps: {},
};

export default PressableIcon;
