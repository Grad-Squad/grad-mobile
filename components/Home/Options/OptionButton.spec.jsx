import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { IconNames } from 'common/Icon/Icon';
import OptionButton from './OptionButton';

it('should match snapshot', () => {
  const optionButton = render(
    <OptionButton label="label" onPress={() => {}} />
  );
  expect(optionButton.toJSON()).toMatchSnapshot();
});

it('should match snapshot with custom iconName', () => {
  const optionButton = render(
    <OptionButton
      label="label"
      iconName={IconNames.question}
      onPress={() => {}}
    />
  );
  expect(optionButton.toJSON()).toMatchSnapshot();
});

it('should call onPress when pressed', () => {
  const onPressMock = jest.fn().mockName('onPressMock');
  const optionButton = render(
    <OptionButton label="label" onPress={onPressMock} />
  );
  fireEvent.press(optionButton.container);
  expect(onPressMock).toHaveBeenCalledTimes(1);
});
