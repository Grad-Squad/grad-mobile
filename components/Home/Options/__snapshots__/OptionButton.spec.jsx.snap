// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 5,
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Text />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Lato_400Regular",
        },
        Object {
          "color": "#3C3C3C",
        },
        Object {
          "fontSize": 25,
          "marginLeft": 5,
        },
      ]
    }
  >
    label
  </Text>
</View>
`;

exports[`should match snapshot with custom iconName 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 5,
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Text />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Lato_400Regular",
        },
        Object {
          "color": "#3C3C3C",
        },
        Object {
          "fontSize": 25,
          "marginLeft": 5,
        },
      ]
    }
  >
    label
  </Text>
</View>
`;
